File: examples/drop.rs
fn skip_first_word(s: &str) -> &str {
    let ws_index = s.chars().take_while(|c| *c != ' ').count();

    &s[ws_index..]
}

fn main() {
    let sentence = String::from("Hello world!");
    let skipped = skip_first_word(&sentence);

    drop(sentence);

    println!("{skipped}");
}

File: drop.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
const char *skip_first_word(const char *str) {
  while (*str && *str != ' ') {
    str++;
  }

  return str;
}

int main() {
  char *sentence = strdup("Hello world!");
  const char *skipped = skip_first_word(sentence);

  free(sentence);

  printf("%s", skipped);
}